index.js

```diff
@@ -3,26 +3,25 @@ const useragent = require('useragent');
 
 const defaults = {
   excludeStatus: [200, 204, 304],
-  excludeResponse: [404],
   tags: ['detailed-response']
 };
 
 const register = (server, options) => {
   options = Hoek.applyToDefaults(defaults, options);
 
-  server.ext('onPreResponse', (request, h) => {
-    const response = request.response;
-    if (!response) {
-      return h.continue;
-    }
+  // log HTTP 301/302 redirects and 404's here:
+  server.events.on({ name: 'response' }, (request) => {
+    // individual routes can disable response logging:
     if (request.route.settings.plugins['hapi-log-response'] && request.route.settings.plugins['hapi-log-response'].enabled === false) {
-      return h.continue;
+      return;
     }
-
-    const statusCode = response.output ? response.output.statusCode : response.statusCode;
-    if (options.excludeStatus.indexOf(statusCode) !== -1) {
-      return h.continue;
+    const statusCode = request.response.statusCode;
+    // exit immediately if not logging anything for this response:
+    if (![301, 302, 404].includes(statusCode) || options.excludeStatus.includes(statusCode)) {
+      return;
     }
+
+    const tags = [].concat(options.tags);
     const data = {
       timestamp: request.info.received,
       id: request.id,
```

Comment:
request.id is now request.info.id

Reply:
I think you can remove labels and instance (hapi doesn't have multiple connections anymore).  

We can also remove timestamp and pid